timemachine/exchange/exchange_mover.py

```diff
@@ -87,108 +95,385 @@ def get_initial_state(water_pdb, mol, ff, seed, nb_cutoff):
+        potentials, params, combined_masses = afe.prepare_host_edge(ff.get_params(), host_config, 0.0)
 
-    bond_list = get_bond_list(host_bps[0].potential)
-    group_idxs = get_group_indices(bond_list, len(combined_masses))
-    barostat_interval = 25
+        host_bps = []
+        for p, bp in zip(params, potentials):
+            host_bps.append(bp.bind(p))
 
-    barostat = MonteCarloBarostat(
-        len(combined_masses), DEFAULT_PRESSURE, temperature, group_idxs, barostat_interval, seed + 1
-    )
+        integrator = LangevinIntegrator(temperature, dt, 1.0, combined_masses, seed)
 
-    # (YTZ): This is disabled because the initial ligand and starting waters are pre-minimized
-    # print("Minimizing the host system...", end="", flush=True)
-    # host_conf = minimize_host_4d([mol], host_config, ff)
-    # print("Done", flush=True)
+        bond_list = get_bond_list(host_bps[0].potential)
+        group_idxs = get_group_indices(bond_list, len(combined_masses))
 
-    host_conf = solvent_conf
-    combined_conf = np.concatenate([host_conf, get_romol_conf(mol)], axis=0)
+        barostat = MonteCarloBarostat(
+            len(combined_masses), DEFAULT_PRESSURE, temperature, group_idxs, barostat_interval, seed + 1
+        )
 
-    ligand_idxs = np.arange(num_water_atoms, num_water_atoms + mol.GetNumAtoms())
+        # (YTZ): This is disabled because the initial ligand and starting waters are pre-minimized
+        # print("Minimizing the host system...", end="", flush=True)
+        # host_conf = minimize_host_4d([mol], host_config, ff)
+        # print("Done", flush=True)
 
-    initial_state = InitialState(
-        host_bps, integrator, barostat, combined_conf, np.zeros_like(combined_conf), solvent_box, 0.0, ligand_idxs
-    )
+        host_conf = solvent_conf
+        combined_conf = np.concatenate([host_conf, get_romol_conf(mol)], axis=0)
+
+        ligand_idxs = np.arange(num_water_atoms, num_water_atoms + mol.GetNumAtoms())
+
+        initial_state = InitialState(
+            host_bps, integrator, barostat, combined_conf, np.zeros_like(combined_conf), solvent_box, 0.0, ligand_idxs
+        )
+
+    else:
+        host_bps, host_masses = openmm_deserializer.deserialize_system(solvent_sys, cutoff=nb_cutoff)
+        integrator = LangevinIntegrator(temperature, dt, 1.0, host_masses, seed)
+        bond_list = get_bond_list(host_bps[0].potential)
+        group_idxs = get_group_indices(bond_list, len(host_masses))
+        barostat_interval = 25
+        barostat = MonteCarloBarostat(
+            len(host_masses), DEFAULT_PRESSURE, temperature, group_idxs, barostat_interval, seed + 1
+        )
+
+        ligand_idxs = [0, 1, 2]  # pick first water molecule to be a "ligand" for targetting purposes
+        initial_state = InitialState(
+            host_bps, integrator, barostat, solvent_conf, np.zeros_like(solvent_conf), solvent_box, 0.0, ligand_idxs
+        )
```

Comment:
nit: clarity -- some partially redundant definitions before, after, and within this branch, might move common definitions outside of branch where possible.
* within branch: define potentials, masses,  `ligand_idxs`
* after branch: construct integrator, barostat, initial state
* remove: second definition of `barostat_interval = 25`

Reply:
willfix